adaptive_memory_v4.0.py

```
@@ -0,0 +1,1551 @@
+                lambda: self.model.encode(
+                    texts, normalize_embeddings=True, show_progress_bar=False
+                ),
+            )
+            return [np.array(e, dtype=np.float32) for e in embeddings]
+        except Exception as e:
+            logger.error(f"Local batch embedding error: {e}")
+            return [None] * len(texts)
+
+
+class OpenAICompatibleEmbeddingProvider(EmbeddingProvider):
+    def __init__(self, api_url: str, api_key: str, model_name: str):
```

Ruff (0.14.10)

Missing return type annotation for special method `__init__`

Add return type annotation: `None`

(ANN204)